apps/web/src/common/index.ts

```diff
@@ -50,7 +50,15 @@ import { ZipFile } from "../utils/streams/zip-stream";
 export const CREATE_BUTTON_MAP = {
   notes: {
     title: strings.addItem("note"),
-    onClick: () => useEditorStore.getState().newSession()
+    onClick: () => {
+      const { sessions, newSession } = useEditorStore.getState();
+      if (sessions.some((session) => "note" in session === false)) {
+        const invalidSession = sessions.find((session) => !("note" in session));
+        useEditorStore.getState().activateSession(invalidSession?.id);
```

Comment:
Let's handle this inside `newSession` in `editor-store`.

Reply:
So you mean like this?

```
  newSession = () => {
    const state = useEditorStore.getState()
    if (state.sessions.some((session) => !("note" in session))) {
      const invalidSession = state.sessions.find((session) => !("note" in session));
      this.activateSession(invalidSession?.id);
    } else {
      this.addSession({
        type: "new",
        id: getId(),
        context: useNoteStore.getState().context,
        saveState: SaveState.NotSaved
      });
    }
  };
```

Reply:
yep. Although you should fix this too:

```diff
- const invalidSession = state.sessions.find((session) => !("note" in session));
+ const session = state.sessions.find((session) => session.type === "new");
```